parquet-thrift/src/main/java/parquet/thrift/projection/FieldProjectionFilter.java

```diff
@@ -18,17 +18,21 @@
  */
 package parquet.thrift.projection;
 
+import java.util.HashSet;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Set;
 
 /**
  * Filter thrift attributes using glob syntax.
  *
  * @author Tianshuo Deng
  */
-public class FieldProjectionFilter {
+public class FieldProjectionFilter implements ProjectionFilter {
   public static final String PATTERN_SEPARATOR = ";";
   List<PathGlobPatternStatus> filterPatterns;
+    private final Set<String> matchedPaths = new HashSet<String>();
+    private final Set<String> unmatchedPaths = new HashSet<String>();
```

Comment:
Do we expect paths to have duplicates? It seems like a List would work just as well here.